Batch insert correction

kebian opened the pull request:
No longer uses the first element to define the values of all inserting records.  This was a bug because if one of the parameters was an expression then the same expression value would be inserted for all records.

As an example of the bug, the following code would create 2 records with the **my_string** field set to **HELLO** (uppercase) when in fact the second record should have its field set to **hello** (lowercase).

``` php
    $test_string = 'Hello';
    $records = [
        [
            'my_string' => DB::raw("upper('$test_string')"),
            'my_int' => 1,
        ],
        [
            'my_string' => DB::raw("lower('$test_string')"),
            'my_int' => 2,
        ],
    ];
    DB::table('my_table')->insert($records);
```

GrahamCampbell commented on src/Illuminate/Database/Query/Grammars/Grammar.php:
```diff
@@ -624,12 +624,10 @@ public function compileInsert(Builder $query, array $values)
-
-		$value = array_fill(0, count($values), "($parameters)");
+		$value = [];
+		foreach($values as &$record) {
```
cs

GrahamCampbell commented:
Could you include tests please?

taylorotwell commented on src/Illuminate/Database/Query/Grammars/Grammar.php:
```diff
@@ -624,12 +624,12 @@ public function compileInsert(Builder $query, array $values)
+		$value = array();
 
-		$value = array_fill(0, count($values), "($parameters)");
+		foreach($values as &$record)
```
Why do you use a reference variable here?

kebian replied:
I often use them to save copying memory around unnecessarily but I agree that it's unnecessary here and will remove it.

GrahamCampbell commented:
Test methods MUST be separated by **2** blank lines, not 1. Please squash to one commit too.

kebian commented:
Other than the page below, is there anywhere I can read up on these coding standards to save us some time?
http://laravel.com/docs/5.0/contributions

I have very limited experience forking and contributing to large projects such as this one.

GrahamCampbell commented:
Sorry, there's not. Our code is self documenting it's cs. This situation will be improved when we move to PSR-2 at the end of May though. :)

kebian commented:
No probs - thanks :)

kebian commented:
From what I understand, to squash these I'll have to rebase.  Is that ok?

kebian committed:
Batch insert correction

No longer uses the first element to define the values of all inserting
records.  This was a bug because if one of the parameters was an expression then
the same expression value would be inserted for all records.

Test included.

lukasgeiter commented:
@kebian Here's a good guide on how to do it: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

kebian commented:
@lukasgeiter Thanks.  I think that got it.
